Spokestack/SpokestackSpeechRecognizer.swift

```diff
@@ -0,0 +1,199 @@
+            }
+        }
+        self.task?.receive() { result in
+            self.handle(result, handleResult: { r in
+                if r.status != "ok" {
+                    self.context.error = SpeechPipelineError.illegalState("Spokestack ASR could not start because its status was \(r.status).")
+                    self.context.dispatch(.error)
+                }
+            })
+        }
+    }
+
+    private func stream(_ frame: Data) {
+        /// - TODO: implement a frame chunking policy to maximize MTU utilization. Should be able to chunk ~50ms of frame data at 16000khz. That's ~1500 bytes, but with TLS and websocket overhead the largest buffer without fragmentation is ~1400 bytes. It would probably make more sense to set the buffer size as a multiple of the audio frame size, so something like 1280 bytes.
+        self.activation += self.configuration.frameWidth
+
+        self.task?.send(URLSessionWebSocketTask.Message.data(frame)) { error in
+            if let error = error {
+                self.context.error = error
+                self.context.dispatch(.error)
+            }
+        }
+        self.task?.receive(completionHandler: self.receive)
+    }
+    
+    private func receive(result: (Result<URLSessionWebSocketTask.Message, Error>)) {
+        self.handle(result, handleResult: { r in
+            if let hypothesis = r.hypotheses.last {
+                self.context.confidence = hypothesis.confidence
+                self.context.transcript = hypothesis.transcript
+                if r.final {
+                    self.context.dispatch(.recognize)
+                }
+            }
+        })
+    }
+
+    private func handle(_ result: Result<URLSessionWebSocketTask.Message, Error>, handleResult: (ASRResult) -> Void) {
+        switch result {
+        case .failure(let error):
+            self.context.error = error
+            self.context.dispatch(.error)
+        case .success(let message):
+            switch message {
+            case .string(let json):
+                do {
+                    guard let jsonData = json.data(using: .utf8) else {
+                        throw SpeechPipelineError.invalidResponse("Could not desearialize the ASR response.")
+                    }
+                    let r = try self.decoder.decode(ASRResult.self, from: jsonData)
+                    if let error = r.error {
+                        throw SpeechPipelineError.failure("Spokestack ASR responded with an error: \(error)")
+                    } else {
+                        handleResult(r)
+                    }
+                } catch let error {
+                    self.context.error = error
+                    self.context.dispatch(.error)
+                }
+            case _:
+                self.configuration.delegateDispatchQueue.async {
+                    self.context.error = SpeechPipelineError.illegalState("Spokestack ASR response with something unknown: \(message)")
```

Comment:
```suggestion
                    self.context.error = SpeechPipelineError.illegalState("unknown response from Spokestack ASR: \(message)")
```
...have you seen a non-string response? Seems odd, and like something that could be handled by a catch or guard instead of normalized in a `switch` statement since there are only 2 cases, 'expected' and 'wtf'.